### PR TITLE
Improved logging invalid form hashes

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I reviewed implemented changes.

#### Why is this the best possible solution? Were any other approaches considered?
Currently thanks to https://github.com/getodk/collect/pull/4906/commits/bba1a981199b0974b964ffa5e33763efd7b1d333 we know that sometimes form details we receive from a server contain invalid hashes. Unfortunately we don't know if it comes from Central or maybe other servers. If it's a bug in Central we need to fix it so I added logging host in a similar way we log server configuration.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It doesn't require testing.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java [hunk 1]
```diff
@@ -241,7 +241,7 @@ private AnalyticsEvents() {
     public static final String ACCURACY_THRESHOLD_DEFAULT = "AccuracyThresholdDefault";
 
     /**
-     * Tracks how often form details with null or empty hashes are provided by a server
+     * Tracks how often form details with invalid hashes are provided by a server
      */
-    public static final String NULL_OR_EMPTY_FORM_HASH = "NullOrEmptyFormHash";
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java:246]: We definitely want to stop logging this?

File: collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsUtils.java [hunk 2]
```diff
@@ -49,6 +50,21 @@ public static void logServerConfiguration(Analytics analytics, String url) {
         String upperCaseURL = url.toUpperCase(Locale.ENGLISH);
         String scheme = upperCaseURL.split(":")[0];
 
+        String urlHash = Md5.getMd5Hash(new ByteArrayInputStream(url.getBytes()));
+        analytics.logEvent(SET_SERVER, scheme + " " + getHostFromUrl(url), urlHash);
+    }
+
+    public static void logInvalidFormHash(String url) {
+        Analytics.log(INVALID_FORM_HASH, "host", getHostFromUrl(url));
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsUtils.java:58]: We'll need to add this as a custom dimension in Firebase.

File: collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java:246]: I just turned it into `INVALID_FORM_HASH` but it logs the same case just with more details.

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java:246]: Ach sorry I meant to delete this comment once I'd got through the review.

**Comment by seadowg** [comment]
@lognaturel did you have any other thoughts on this? If not, I'm happy with it.

**Comment by lognaturel** [comment]
Thanks for pinging me! I wanted to give this another look because on first pass I thought I had seen an issue but it after all I think it's good. 👍

**Review by lognaturel — APPROVED** [review]

**Pull request merged**